test/topology/test_topology.py

```diff
@@ -99,15 +99,31 @@ async def no_joining_nodes():
 async def test_replace_different_ip(manager: ManagerClient, random_tables) -> None:
     servers = await manager.running_servers()
     await manager.server_stop(servers[0].server_id)
-    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = False)
+    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = False, use_host_id = False)
+    await manager.server_add(replace_cfg)
+    # TODO: check that group 0 no longer contains the replaced node (#12153)
+
+@pytest.mark.asyncio
+async def test_replace_using_host_id_different_ip(manager: ManagerClient, random_tables) -> None:
+    servers = await manager.running_servers()
+    await manager.server_stop(servers[0].server_id)
+    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = False, use_host_id = True)
     await manager.server_add(replace_cfg)
     # TODO: check that group 0 no longer contains the replaced node (#12153)
 
 @pytest.mark.asyncio
 async def test_replace_reuse_ip(manager: ManagerClient, random_tables) -> None:
     servers = await manager.running_servers()
     await manager.server_stop(servers[0].server_id)
-    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = True)
+    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = True, use_host_id = False)
+    await manager.server_add(replace_cfg)
+    # TODO: check that group 0 no longer contains the replaced node (#12153)
+
+@pytest.mark.asyncio
+async def test_replace_using_host_id_reuse_ip(manager: ManagerClient, random_tables) -> None:
+    servers = await manager.running_servers()
+    await manager.server_stop(servers[0].server_id)
+    replace_cfg = ReplaceConfig(replaced_id = servers[0].server_id, reuse_ip_addr = True, use_host_id = True)
```

Comment:
Maybe we could do the two replaces (using host ID and not using host ID) in a single test, so we don't grow the number of tests cases? Each test case requires starting a separate cluster - which takes some time.

(Maybe we should do the same with `reuse_ip_addr` - but out of this PR's scope)

Reply:
okay.

Maybe we should take the same approach in
https://github.com/scylladb/scylla-dtest/pull/2946

@fruch what do you think?

Reply:
@kbr-scylla this has proven to be more difficult than I thought, since it the second replace in a row fails (ta least some of the times) because it sees one of the nodes as down.  I think it's the first replacing node that is not recognized as UN by all other nodes when add_server resolves.

It's out of scope for this PR to deal with that, so I'll keep the tests separated and we can merge them as a follow up.

Reply:
Ok. Maybe I'll work on merging them later.